options/core/src/main/java/polanski/option/OptionAssertion.java

```diff
@@ -0,0 +1,63 @@
+package polanski.option;
+
+import android.support.annotation.NonNull;
+
+import polanski.option.function.Func1;
+
+public final class OptionAssertion<T> {
+
+    @NonNull
+    private final Option<T> actual;
+
+    OptionAssertion(@NonNull final Option<T> actual) {
+        this.actual = actual;
+    }
+
+    public void assertIsNone() {
+        if (!actual.isNone()) {
+            fail("Option was not None");
+        }
+    }
+
+    public OptionAssertion<T> assertIsSome() {
+        if (!actual.isSome()) {
+            fail("Option was not Some");
+        }
+        return this;
+    }
+
+    public OptionAssertion<T> assertValue(Func1<T, Boolean> predicate) {
+        if (!actual.isSome()) {
+            fail("Option was not Some");
+        }
+
+        if (!actual.filter(predicate).isSome()) {
+            fail("Option value did not match predicate");
+        }
+        return this;
+    }
+
+    public OptionAssertion<T> assertValue(T expected) {
```

Comment:
I should add null checks for the value - I guess? I mean the value could be null, but the test should use `assertNone` instead.